### PR TITLE
[LWG motion 20] P2757R3 Type checking format args

### LINKED ISSUE #6306: [2023-06 LWG Motion 20] P2757R3 Type checking format args

### PR DESCRIPTION
**Pull request by burblebee**
Fixes #6306.

### CODE REVIEW

File: source/utilities.tex [hunk 1]
```diff
@@ -16301,6 +16310,68 @@
...
+Each type in \tcode{Ts...} is one of
+\keyword{bool},
+\tcode{char_type},
+\keyword{int},
+\keyword{unsigned} \keyword{int},
+\keyword{long} \keyword{long} \keyword{int},
+\keyword{unsigned} \keyword{long} \keyword{long} \keyword{int},
+\keyword{float},
+\keyword{double},
+\keyword{long} \keyword{double},
```
Comment (JohelEGP) [inline source/utilities.tex:16332]:
```suggestion
\tcode{\keyword{unsigned} \keyword{int}},
\tcode{\keyword{long} \keyword{long} \keyword{int}},
\tcode{\keyword{unsigned} \keyword{long} \keyword{long} \keyword{int}},
\keyword{float},
\keyword{double},
\tcode{\keyword{long} \keyword{double}},
```

File: source/utilities.tex [hunk 2]
```diff
@@ -16301,6 +16310,68 @@
...
+\remarks
+Call expressions where
+\tcode{id >= num_args_} or
+the type of the corresponding format argument
+(after conversion to \tcode{basic_format_arg<Context>})
+is not one of the types in \tcode{Ts...}
+are not core constant expressions\iref{expr.const}.
+\end{itemdescr}
+
+\indexlibrarymember{check_dynamic_spec}{basic_format_parse_context}%
```
Comment (JohelEGP) [inline source/utilities.tex:16347]:
```suggestion
\indexlibrarymember{check_dynamic_spec_integral}{basic_format_parse_context}%
```

File: source/utilities.tex [hunk 3]
```diff
@@ -16301,6 +16310,68 @@
...
+\begin{itemdescr}
+\pnum
+\effects
+Equivalent to:
+\begin{codeblock}
+check_dynamic_spec<int, unsigned int, long long int, unsigned long long int>(id);
+\end{codeblock}
+\end{itemdescr}
+
+\indexlibrarymember{check_dynamic_spec}{basic_format_parse_context}%
```
Comment (JohelEGP) [inline source/utilities.tex:16361]:
```suggestion
\indexlibrarymember{check_dynamic_spec_string}{basic_format_parse_context}%
```

**Review by JohelEGP — COMMENTED** [review]
I'm using <https://isocpp.org/files/papers/P2757R3.html>.

File: source/utilities.tex (hunk 1 above)
Comment (burblebee) [reply source/utilities.tex:16332]: done, but why bother? because the spacing is slightly different?

**Review by JohelEGP — APPROVED** [review]

File: source/utilities.tex (hunk 1 above)
Comment (JohelEGP) [reply source/utilities.tex]: It better encodes what is meant.

**Pull request merged**